apiv2/project/project.go

```diff
@@ -523,55 +493,53 @@ func (c *RESTClient) DeleteProjectMetadataValue(ctx context.Context, p *modelv2.
 }
 
 // ListProjectRobots returns a list of all robot accounts in project p.
-func (c *RESTClient) ListProjectRobots(ctx context.Context, p *modelv2.Project) ([]*model.RobotAccount, error) {
+func (c *RESTClient) ListProjectRobots(ctx context.Context, p *modelv2.Project) ([]*modelv2.Robot, error) {
 	if p == nil {
 		return nil, &ErrProjectNotProvided{}
 	}
 
-	resp, err := c.LegacyClient.Products.GetProjectsProjectIDRobots(
-		&products.GetProjectsProjectIDRobotsParams{
-			ProjectID: int64(p.ProjectID),
-			Context:   ctx,
-		}, c.AuthInfo)
+	resp, err := c.V2Client.Robotv1.ListRobotV1(&robotv1.ListRobotV1Params{
+		ProjectNameOrID: strconv.Itoa(int(p.ProjectID)),
```

Comment:
You do this conversion quite often. Maybe you should append a `ProjectIDAsString()` method to the `Project` struct?

```go
func (m *Project) ProjectIDAsString() string {
	return strconv.Itoa(int(m.ProjectID))
}
```

Reply:
As `*modelv2.Project` is not a local type we cannot use it as receiver like you described.
I've introduced a function that takes the `int32` projectID as argument instead.